Sources/Utils/Box.swift

```diff
@@ -1,33 +1,39 @@
-/// An arbitrary container which stores a **constant** value of type `T`.
+/// An arbitrary container (and property wrapper) which stores a **mutable** value of type `T`.
 ///
-/// This main purpose of this object is to encapsulate value types so that they can be used like a reference type
-/// (e.g. pass value types around without copying them, "share" value types between closures, etc)
+/// The main purpose of this object is to encapsulate value types so that they can be used like a reference type
+/// (e.g. pass value types around without copying them, memoize value types inside closures via capture list, etc)
+@propertyWrapper
+@dynamicMemberLookup
 public final class Box<T> {
 
-    /// The encapsulated value.
-    public let value: T
+    /// The wrapped value.
+    public var wrappedValue: T
 
-    /// Instantiate a new constant value box with the given value.
+    /// Instantiate a new mutable value box with the given value.
     ///
-    /// - parameter value: the value to encapsulate.
+    /// - parameter wrappedValue: the value to encapsulate.
     ///
     /// - returns: a newly instantiated box with the encapsulated value.
-    public init(_ value: T) { self.value = value }
+    public init(wrappedValue: T) { self.wrappedValue = wrappedValue }
+
+    public subscript<U>(dynamicMember keyPath: KeyPath<T, U>) -> U { wrappedValue[keyPath: keyPath] }
 }
 
-/// An arbitrary container which stores a **variable** value of type `T`.
-///
-/// This main purpose of this object is to encapsulate value types so that they can be used like a reference type
-/// (e.g. pass value types around without copying them, "share" value types between closures, etc)
-public final class VarBox<T> {
+extension Box {
 
-    /// The encapsulated value.
-    public var value: T
+    /// The wrapped value (compact).
+    public var value: T {
+        get { wrappedValue }
+        set { wrappedValue = newValue }
+    }
```

Comment:
it appears that internally only Network+URLSessionNetworkStack.swift used Box, are we keeping this `value` so that it doesn't break existing codebases that rely on alicerce? if so, won't it also be confusing exposing `wrappedValue` as well? maybe `wrappedValue` could be private, keeping older codebases compliant but not obfuscating the internals of what it actually does (wrapping a value with a propertyWrapper).

Or maybe i'm just over-analyzing all of this 😅

Reply:
> are we keeping this value so that it doesn't break existing codebases that rely on alicerce? 
Yes, the main motivation is to make the transition smoother. Also, `Box<T>` can now be used both as a PW and as a regular type, so when used as a regular type I thought it would be a bit more ergonomic to have a `value` property, as well as an `init` without the parameter name.

> if so, won't it also be confusing exposing `wrappedValue` as well?
I don't think this will be confusing, as `wrappedValue` is well known to be tightly coupled to PWs. Also, users can always look into the code to see what's going on, right? We even have documentation (e.g. `(compact)`) in place. 🤓 

> maybe `wrappedValue` could be private, keeping older codebases compliant but not obfuscating the internals of what it actually does (wrapping a value with a propertyWrapper).
Sadly this doesn't work, because `wrappedValue` is a requirement of PWs, and must have the same access level as its enclosing PW. A similar thing happens with the `init(wrappedValue: T)`

![Screen Shot 2022-01-18 at 09 22 09](https://user-images.githubusercontent.com/1391324/149908092-05ab1049-47f5-41af-af22-b86bc9a64533.png)

Hope this answers your questions 😄

Reply:
it does, thanks! 🙇